diff --git a/scripts/setlocalversion b/scripts/setlocalversion
index 8fcc5ce..fc0dd2c 100755
--- a/scripts/setlocalversion
+++ b/scripts/setlocalversion
@@ -77,7 +77,7 @@ scm_version()
 				printf '%s%s%s' -g "`git show-ref -s --abbrev --tags $tag 2>/dev/null`" $commit
 			# If we don't have a tag at all we print -g{commitish}.
 			else
-				printf '%s%s' -g $head
+				printf '%s%s'
 			fi
 		fi
 
@@ -91,7 +91,7 @@ scm_version()
 
 		# Check for uncommitted changes
 		if git diff-index --name-only HEAD | grep -qv "^scripts/package"; then
-			printf '%s' -dirty
+			printf '%s'
 		fi
 
 		# All done with git
